app/Metadata/GitHubFunctions.php

```diff
@@ -2,40 +2,30 @@
 
 namespace App\Metadata;
 
-use App\Exceptions\NotInCacheException;
-use App\Exceptions\NotMasterException;
+use App\Exceptions\VersionControlException;
 use App\Facades\Helpers;
 use App\ModelFunctions\JsonRequestFunctions;
 use App\Models\Configs;
-use Exception;
 use Illuminate\Support\Facades\Config;
 
 class GitHubFunctions
 {
-	public string $head;
-
-	public string $branch;
-
 	private GitRequest $gitRequest;
+	protected string $head;
+	protected string $branch;
 
 	/**
 	 * Base constructor.
 	 *
 	 * @param GitRequest $gitRequest
+	 *
+	 * @throws VersionControlException
 	 */
 	public function __construct(GitRequest $gitRequest)
 	{
 		$this->gitRequest = $gitRequest;
-		try {
-			$this->branch = $this->get_current_branch();
-			$this->head = $this->get_current_commit();
-			// @codeCoverageIgnoreStart
-			// when testing on master branch this is not covered.
-		} catch (Exception $e) {
-			$this->branch = false;
-			$this->head = false;
-		}
```

Comment:
Removing this causes exceptions in testing (https://github.com/LycheeOrg/Lychee/pull/1274#issuecomment-1107869076)